src/overlays/actors/ovl_En_Bjt/z_en_bjt.c

```diff
@@ -31,49 +53,408 @@ const ActorInit En_Bjt_InitVars = {
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_HIT1,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 10, 68, 0, { 0, 0, 0 } },
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit2 D_80BFDF74 = { 0, 0, 0, 0, MASS_IMMOVABLE };
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
+
+typedef enum {
+    /* -1 */ TOILET_HAND_ANIM_NONE = -1,
+    /*  0 */ TOILET_HAND_ANIM_WAITING,
+    /*  1 */ TOILET_HAND_ANIM_WAITING_MORPH,  // Same as TOILET_HAND_ANIM_WAITING, but with morph frames
+    /*  2 */ TOILET_HAND_ANIM_WAGGING_FINGER, // Wrong
+    /*  3 */ TOILET_HAND_ANIM_THUMBS_UP,      // Right
+    /*  4 */ TOILET_HAND_ANIM_OPEN_HAND,
+    /*  5 */ TOILET_HAND_ANIM_FIST // i.e. holding the reward
+} ToiletHandAnimations;
+
+static AnimationInfoS sAnimationInfos[] = {
+    /* 0 */ { &gToiletHandWaitingAnim, 1.0f, 0, -1, 0, 0 },
+    /* 1 */ { &gToiletHandWaitingAnim, 1.0f, 0, -1, 0, -4 },
+    /* 2 */ { &gToiletHandWaggingFingerAnim, 1.0f, 0, -1, 0, -4 },
+    /* 3 */ { &gToiletHandThumbsUpAnim, 1.0f, 0, -1, 0, -4 },
+    /* 4 */ { &gToiletHandOpenHandAnim, 1.0f, 0, -1, 0, -4 },
+    /* 5 */ { &gToiletHandFistAnim, 1.0f, 0, -1, 0, -4 },
+};
+
+void EnBjt_UpdateAnimation(EnBjt* this) {
+    this->skelAnime.playSpeed = this->animPlaySpeed;
+    SkelAnime_Update(&this->skelAnime);
+}
+
+s32 EnBjt_ChangeAnimation(EnBjt* this, s32 animIndex) {
+    s32 changeAnim = false;
+    s32 changed = false;
+
+    if ((animIndex == TOILET_HAND_ANIM_WAITING) || (animIndex == TOILET_HAND_ANIM_WAITING_MORPH)) {
+        if (!((this->curAnimIndex == TOILET_HAND_ANIM_WAITING) ||
+              (this->curAnimIndex == TOILET_HAND_ANIM_WAITING_MORPH))) {
+            changeAnim = true;
+        }
+    } else if (this->curAnimIndex != animIndex) {
+        changeAnim = true;
+    }
+
+    if (changeAnim) {
+        this->curAnimIndex = animIndex;
+        changed = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfos, animIndex);
+        this->animPlaySpeed = this->skelAnime.playSpeed;
+    }
+
+    return changed;
+}
+
+void EnBjt_UpdateCollision(EnBjt* this, PlayState* play) {
+    static Vec3f sColliderBasePos = { 0.0f, 8.0f, 10.0f };
+    s32 pad;
+    Vec3f pos;
+    f32 height;
+
+    if (this->stateFlags & (TOILET_HAND_STATE_APPEARING | TOILET_HAND_STATE_VISIBLE)) {
+        Lib_Vec3f_TranslateAndRotateY(&this->actor.world.pos, this->actor.shape.rot.y, &sColliderBasePos, &pos);
+        Math_Vec3f_ToVec3s(&this->collider.dim.pos, &pos);
+        height = this->actor.focus.pos.y - this->actor.world.pos.y;
```

Comment:
Is the height temp needed?

Reply:
Yeah, needed to get it into fv0: https://decomp.me/scratch/wO3vE